common-methods.go

```diff
@@ -57,17 +56,18 @@ func getSource(sourceURL string) (reader io.ReadSeeker, err *probe.Error) {
 }
 
 // putTarget writes to URL from reader. If length=-1, read until EOF.
-func putTarget(targetURL string, reader io.ReadSeeker, size int64) *probe.Error {
+func putTarget(targetURL string, reader io.ReadSeeker, size int64) (int64, *probe.Error) {
```

Comment:
putTarget doesn't need to return size. It needs to retry if possible and fail otherwise.